src/ceph-volume/ceph_volume/api/lvm.py

```diff
@@ -259,7 +259,10 @@ def dmsetup_splitname(dev):
         "--separator=';'", '--nameprefixes', dev
     ]
     out, err, rc = process.call(command)
-    return _splitname_parser(out)
+    if len(out) != 1:
+        logger.error('Unexpected "dmsetup splitname" output for "%s"', dev)
+        return {}
```

Comment:
this function doesn't to any parsing, it would be unexpected to produce a dictionary here

Reply:
@alfredodeza we have only two choices print an error and exit or assume this is not a device mapper volume device and return a dict.

Or maybe I didn't get it and you just what I return the dict withall expected keys and empty values

Reply:
I mean, this function is not in charge of parsing, that should be delegated to the one that parses.

The return value is determined by the parser, not this function. The empty dict can be returned by the parser

Reply:
oh my bad, I misreading you, thank for the rephrasing :) I will make the change.